chain/near/src/data_source.rs

```diff
@@ -75,6 +75,13 @@ impl blockchain::DataSource<Chain> for DataSource {
         self.source.start_block
     }
 
+    fn has_expired(&self, block: BlockNumber) -> bool {
+        match self.source.end_block {
```

Comment:
if you add end_block() to the trait, then this could be implemented in the trait itself as default and wouldn't need to be copied into all the trait implementations. Might not be a bad trade-off

Reply:
@mangas then we'd still have to go around implementing `end_block()` on all Datasources right? its either `end_block()` or `has_expired()`. Lets go with `has_expired` then?